coil-gif/src/main/java/coil/transform/AnimatedTransformation.kt

```diff
@@ -0,0 +1,31 @@
+package coil.transform
+
+import android.graphics.Canvas
+import coil.size.Size
+import android.graphics.PixelFormat as AndroidPixelFormat
+
+/**
+ * An interface for applying transformation on GIFs, animated WebPs, and animated HEIFs.
+ */
+interface AnimatedTransformation {
+
+    /**
+     * Apply transformation on [canvas]
+     *
+     * Note: Do not allocate objects in this method as it will be invoked for each frame of the animation.
+     *
+     * @return Opacity of the result after drawing.
+     * @see AndroidPixelFormat
+     */
+    fun transform(canvas: Canvas, size: Size): PixelFormat
```

Comment:
Thinking about this some more, I don't think we need to pass `size` in here. In regular `Transformations` it's used so you can optionally return a bitmap with a different size. That isn't possible with animated transformations as they interact directly with the canvas. Let's change this to `fun transform(canvas: Canvas): PixelFormat`.

Reply:
Even I was thinking the same.